### PR TITLE
Update raspi-config.md

### PR DESCRIPTION
**Pull request by JamesH65**
Update to latest version of raspi-config

### CODE REVIEW

**Comment by JamesH65** [comment]
@andrum99 for review

File: configuration/raspi-config.md [hunk 1]
```diff
@@ -56,42 +54,69 @@ Generally speaking, `raspi-config` aims to provide the functionality to make the
 <a name="menu-options"></a>
 ## Menu options
 
+### System Options
+
+The systems optiosn submenu allows you to make configuration changes to various parts of the boot, login and networking process, along with some other system level changes.
```
Comment (lurch) [inline configuration/raspi-config.md:59]: `systems optiosn` -> `system options`

File: configuration/raspi-config.md [hunk 2]
```diff
@@ -56,42 +54,69 @@ Generally speaking, `raspi-config` aims to provide the functionality to make the
...
 <a name="hostname"></a>
 #### Hostname
 
 Set the visible name for this Pi on a network.
 
 <a name="boot-options"></a>
-### Boot Options
+#### Boot / Auto login
 
 From here you can change what happens when your Pi boots. Use this option to change your boot preference to command line or desktop. You can choose whether boot-up waits for the network to be available, and whether the Plymouth splash screen is displayed at boot-up.
```
Comment (lurch) [inline configuration/raspi-config.md:82]: "choose whether boot-up waits for the network to be available" - that's controlled by the next option, not by this option.

"whether the Plymouth splash screen is displayed" - this option doesn't seem to be there?

The last sentence here should probably say something like "You can choose to have your Pi boot to a login prompt, or whether to log in as the `pi` user automatically."

File: configuration/raspi-config.md [hunk 3]
```diff
@@ -56,42 +54,69 @@ Generally speaking, `raspi-config` aims to provide the functionality to make the
...
+
+### Display Options
+
+<a name="resolution"></a>
+#### Resolution
+
+Define the default HDMI/DVI video resolution to use when the system boots without a TV or monitor being connected. This can have an effect on RealVNC if the VNC option is enabled.
+
+<a name="overscan"></a>
+#### Overscan
```
Comment (lurch) [inline configuration/raspi-config.md:104]: On my version of `raspi-config`, this menu item is named "Underscan" ?

File: configuration/raspi-config.md [hunk 4]
```diff
@@ -56,42 +54,69 @@ Generally speaking, `raspi-config` aims to provide the functionality to make the
...
+Any changes will take effect after a reboot. You can have greater control over the settings by editing [config.txt](config-txt/README.md).
+
+On some displays, particularly monitors, disabling overscan will make the picture fill the whole screen and correct the resolution. For other displays, it may be necessary to leave overscan enabled and adjust its values.
+
+<a name="pixel-doubling"></a>
+#### Pixel Doubling
+
+Enable/disable 2x2 pixel mapping.
+
+#### Screen Blanking
```
Comment (lurch) [inline configuration/raspi-config.md:117]: My version of `raspi-config` has a "Composite Video" menu option between the "Pixel Doubling" and "Screen Blanking" options?

File: configuration/raspi-config.md [hunk 5]
```diff
@@ -135,51 +160,59 @@ Enable/disable shell and kernel messages on the serial connection.
...
+#### Remote GPIO
+
+Enable or disable remote access to the GPIO pins.
+
+### Performance Options
+
 <a name="overclock"></a> 
 ### Overclock
 
 It is possible to overclock your Raspberry Pi's CPU. The default is 700MHz but it can be set up to 1000MHz. The overclocking you can achieve will vary; overclocking too high may result in instability. Selecting this option shows the following warning:
```
Comment (lurch) [inline configuration/raspi-config.md:172]: "The default is 700MHz" - nope, the default varies depending on the model of Pi you're using :wink:

File: configuration/raspi-config.md [hunk 6]
```diff
@@ -135,51 +160,59 @@ Enable/disable shell and kernel messages on the serial connection.
...
 
-If you have installed Raspberry Pi OS using NOOBS, the filesystem will have been expanded automatically. There may be a rare occasion where this is not the case, e.g. if you have copied a smaller SD card onto a larger one. In this case, you should use this option to expand your installation to fill the whole SD card, giving you more space to use for files. You will need to reboot the Raspberry Pi to make this available. Note that there is no confirmation: selecting the option begins the partition expansion immediately.
+Enable or disable a read-only filesystem
 
-<a name="overscan"></a>
-#### Overscan
+#### Fan
 
-Old TV sets had a significant variation in the size of the picture they produced; some had cabinets that overlapped the screen. TV pictures were therefore given a black border so that none of the picture was lost; this is called overscan. Modern TVs and monitors don't need the border, and the signal doesn't allow for it. If the initial text shown on the screen disappears off the edge, you need to enable overscan to bring the border back.
+Set the bahaviour of a GPIO connected fan
```
Comment (lurch) [inline configuration/raspi-config.md:187]: `bahaviour` -> `behaviour`

File: configuration/raspi-config.md [hunk 7]
```diff
@@ -135,51 +160,59 @@ Enable/disable shell and kernel messages on the serial connection.
...
-<a name="advanced-options"></a>
-### Advanced Options
+Change the amount of memory made available to the GPU.
 
-<a name="expand-filesystem"></a>
-#### Expand Filesystem
+#### Overlay File System
 
-If you have installed Raspberry Pi OS using NOOBS, the filesystem will have been expanded automatically. There may be a rare occasion where this is not the case, e.g. if you have copied a smaller SD card onto a larger one. In this case, you should use this option to expand your installation to fill the whole SD card, giving you more space to use for files. You will need to reboot the Raspberry Pi to make this available. Note that there is no confirmation: selecting the option begins the partition expansion immediately.
+Enable or disable a read-only filesystem
```
Comment (lurch) [inline configuration/raspi-config.md:183]: This also allows you to set the boot partition read-only or not?

File: configuration/raspi-config.md [hunk 8]
```diff
@@ -201,8 +234,20 @@ Enable/disable the experimental OpenGL Fake KMS desktop graphics driver.
...
+
+Enable/Display the xcompmgr composition manager
+
+#### Network Interface Names
+
+Enable or disable predictable network interface names. 
+
+#### Network Proxy Settings
+
+Configure the networks proxy settings.
```
Comment (lurch) [inline configuration/raspi-config.md:247]: `networks` should either be `network` or `network's`

File: configuration/raspi-config.md [hunk 9]
```diff
@@ -201,8 +234,20 @@ Enable/disable the experimental OpenGL Fake KMS desktop graphics driver.
...
+Enable/Display the xcompmgr composition manager
+
+#### Network Interface Names
+
+Enable or disable predictable network interface names. 
+
+#### Network Proxy Settings
+
+Configure the networks proxy settings. 
+
```
Comment (lurch) [inline configuration/raspi-config.md:248]: The "Advanced Settings" menu also has options for "Boot Order" and "Bootloader version".

**Comment by lurch** [comment]
Also, I only see "Legacy" and "GL (Fake KMS)" listed under the "GL driver" menu, there's not a "GL (Full KMS)" option?

**Comment by lurch** [comment]
ping also @spl237 in case he has any comments.

File: configuration/raspi-config.md (hunk 9 above)
Comment (JamesH65) [reply configuration/raspi-config.md:248]: Hmm, not on mine - might need to do an update.

File: configuration/raspi-config.md (hunk 7 above)
Comment (JamesH65) [reply configuration/raspi-config.md:183]: No idea. I just continued on with what the menu says.

File: configuration/raspi-config.md (hunk 5 above)
Comment (JamesH65) [reply configuration/raspi-config.md:172]: Ah, missed that!

**Comment by spl237** [comment]
> Also, I only see "Legacy" and "GL (Fake KMS)" listed under the "GL driver" menu, there's not a "GL (Full KMS)" option?

You'll only see the full KMS option on Pi 3 and below, IIRC.

File: configuration/raspi-config.md (hunk 4 above)
Comment (JamesH65) [reply configuration/raspi-config.md:117]: Doesn't appear on Pi4 I have here...

**Comment by lurch** [comment]
Ahhh, I didn't realise that `raspi-config` showed different options, depending on what model of Pi you're running it on! I guess that also explains why me and James were seeing different options? And probably makes it a bit tricky to document accurately...
(@JamesH65 I was running it on a Pi400)

File: configuration/raspi-config.md (hunk 7 above)
Comment (lurch) [reply configuration/raspi-config.md:183]: Can probably be ignored then - this is probably one of those options where anyone using it, understands what it does :)

File: configuration/raspi-config.md (hunk 9 above)
Comment (JamesH65) [reply configuration/raspi-config.md:248]: Nope still not there.  Ah, maybe because i am on a Pi3. Will try on a Pi4.

**Comment by spl237** [comment]
FWIW, I wouldn't personally have said it was worth spending that much time documenting raspi-config; the only people using it should be people experienced enough to know their way around a Linux command line; everyone else should be using the desktop GUI. But yes, it does show different options depending on the hardware on which it is running.

File: configuration/raspi-config.md [hunk 10]
```diff
@@ -15,27 +15,25 @@ sudo raspi-config
 
 The `sudo` is required because you will be changing files that you do not own as the `pi` user.
 
-You should see a blue screen with options in a grey box in the centre, like so:
+You should see a blue screen with options in a grey box in the centre, something like so (this might not be exactly what you see):
```
Comment (andrum99) [inline configuration/raspi-config.md:18]: Not sure about 'should'. What about:

`A blue menu screen will be displayed, with options in a grey box:`

<picture>

`Note that the menu shown may differ slightly.`

Other than that, I think AndrewS has spotted everything.

File: configuration/raspi-config.md (hunk 9 above)
Comment (JamesH65) [reply configuration/raspi-config.md:248]: Yup.

**Comment by andrum99** [comment]
> FWIW, I wouldn't personally have said it was worth spending that much time documenting raspi-config; the only people using it should be people experienced enough to know their way around a Linux command line; everyone else should be using the desktop GUI.

Good point.

**Comment by andrum99** [comment]
I suspect this page of the docs is from a time before `rc-gui`, when `raspi-config` was the most user-friendly way to set these options. I wonder if it would be worth mentioning the GUI tool at the top of this page?

**Comment by andrum99** [comment]
> I suspect this page of the docs is from a time before `rc-gui`, when `raspi-config` was the most user-friendly way to set these options. I wonder if it would be worth mentioning the GUI tool at the top of this page?

Never mind - it already does.

**Comment by JamesH65** [comment]
OK, pushed the changes. As @spl237 says, not worth spending too much time on this, but at least now its mostly up to date.

**Comment by lurch** [comment]
FYI I just pushed a couple of typo fixes directly to `master`.

**Pull request merged**